aten/src/ATen/native/cpu/GridSamplerKernel.cpp

```diff
@@ -620,44 +638,43 @@ struct ApplyGridSample<scalar_t, 2, GridSamplerInterpolation::Bilinear,
     integer_t i_gInp_sw_offset_arr[iVec::size()];
     // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
     integer_t i_gInp_se_offset_arr[iVec::size()];
-    i_gInp_nw_offset.store(i_gInp_nw_offset_arr);
-    i_gInp_ne_offset.store(i_gInp_ne_offset_arr);
-    i_gInp_sw_offset.store(i_gInp_sw_offset_arr);
-    i_gInp_se_offset.store(i_gInp_se_offset_arr);
-
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_nw_mask_arr[iVec::size()];
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_ne_mask_arr[iVec::size()];
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_sw_mask_arr[iVec::size()];
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_se_mask_arr[iVec::size()];
-    nw_mask.store(i_nw_mask_arr);
-    ne_mask.store(i_ne_mask_arr);
-    sw_mask.store(i_sw_mask_arr);
-    se_mask.store(i_se_mask_arr);
+    if (input_requires_grad) {
+      auto i_gInp_nw_offset = i_y_n * iVec(inp_W) + i_x_w;
+      auto i_gInp_ne_offset = i_gInp_nw_offset + iVec(1);
+      auto i_gInp_sw_offset = i_gInp_nw_offset + iVec(inp_W);
+      auto i_gInp_se_offset = i_gInp_sw_offset + iVec(1);
+
+      i_gInp_nw_offset.store(i_gInp_nw_offset_arr);
+      i_gInp_ne_offset.store(i_gInp_ne_offset_arr);
+      i_gInp_sw_offset.store(i_gInp_sw_offset_arr);
+      i_gInp_se_offset.store(i_gInp_se_offset_arr);
+    }
 
     // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
     scalar_t gInp_corner_arr[Vec::size()];
 
+    // ----- End of unnecessary variables for input_requires_grad is false.
```

Comment:
not sure what this means?

Reply:
There's a block of variables that ends here and that are basically unnecessary when `input_requires_gradient` is false (but required to be declared as they are used within the if-blocks). I removed this line and changed an earlier comment so that the variables are explicitly named there.